src/ComposerPlugin.php

```diff
@@ -227,7 +227,7 @@ private function detectGitDir(): void
     {
         $path = getcwd();
 
-        while (file_exists($path)) {
+        if (file_exists($path)) {
```

Comment:
This will only check on one level. The former will go all the way to the top level and check for a `.git`-folder. To fix the possible endless loop you would need to check whether `$path` is the top-level and then break the while loop

Reply:
The problem is that `basename('/')` will return `''`  (an empty string) and `basename('')` will still return an empty string. 

But `file_exists('')` should return true and therefore break the chain.

Are you by any chance running this on windows?

Reply:
> This will only check on one level. The former will go all the way to the top level and check for a `.git`-folder. To fix the possible endless loop you would need to check whether `$path` is the top-level and then break the while loop

Ah yes, I understand.



> The problem is that `basename('/')` will return `''` (an empty string) and `basename('')` will still return an empty string.
> 
> But `file_exists('')` should return true and therefore break the chain.
> 
> Are you by any chance running this on windows?

No, I'm using a Mac.

Reply:
`$path` was returning '/' while beeing stuck in the endless loop. So if `$path` does match the `DIRECTORY_SEPERATOR` it will break the chain. This should work on any OS.

Reply:
Right again, we would have to add it above line 236.... that's what I meant anyway *caugh* :P